src/components/implementation/no_interface/scheddev/sl.c

```diff
@@ -51,6 +51,28 @@ sl_cs_exit_contention(union sl_cs_intern *csi, union sl_cs_intern *cached, sched
 	return 0;
 }
 
+/*
+ * @return: 1 if it's already WOKEN.
+ *	    0 if it successfully blocked in this call.
+ */
+int
+sl_thd_block_no_cs(struct sl_thd *t, int is_timeout)
+{
+	assert(t);
+
+	if (unlikely(t->state == SL_THD_WOKEN)) {
+		t->state = SL_THD_RUNNABLE;
+		sl_cs_exit();
```

Comment:
Yes, big mistake. I'll correct it. Thank you. I'll also add an assertion in `sl_cs_exit`.